optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -243,9 +251,21 @@ private void registerVariableAccessor(DescriptorPolicy descriptorPolicy,
                     + "Maybe the annotation is defined on both the field and its getter.");
         }
         if (variableAnnotationClass.equals(PlanningVariable.class)) {
-            GenuineVariableDescriptor<Solution_> variableDescriptor = new GenuineVariableDescriptor<>(this,
-                    memberAccessor);
+            GenuineVariableDescriptor<Solution_> variableDescriptor = new BasicVariableDescriptor<>(
+                    this, memberAccessor);
             declaredGenuineVariableDescriptorMap.put(memberName, variableDescriptor);
+        } else if (variableAnnotationClass.equals(PlanningCollectionVariable.class)) {
+            if (List.class.isAssignableFrom(memberAccessor.getType())) {
+                GenuineVariableDescriptor<Solution_> variableDescriptor = new ListVariableDescriptor<>(
+                        this, memberAccessor);
+                declaredGenuineVariableDescriptorMap.put(memberName, variableDescriptor);
+            } else {
+                throw new IllegalStateException("The entityClass (" + entityClass
+                        + ") has a " + PlanningCollectionVariable.class.getSimpleName()
+                        + " annotated member (" + memberAccessor
+                        + ") that has an unsupported type (" + memberAccessor.getType() + ").\n"
+                        + "A subtype of " + List.class.getCanonicalName() + " is expected.");
```

Comment:
=> "Maybe ..." use the type List instead?

Reply:
I'm leaving the `List.class.isAssignableFrom(memberAccessor.getType())` condition intact for now.

Technically it's possible to have
```java
@PlanningCollectionVariable MyList<Task> tasks;
```

It won't work out of the box because the built-in `SolutionCloner` is dumb and tries to assign an `ArrayList` instance to the `MyList` field. But I believe that it's possible to use the above is user provides a custom `SolutionCloner`.

Reply:
For example, `ArrayList` and `LinkedList` work as a list variable types out of the box (thanks to the support from the built-in SolutionCloner).